gatling-core/src/main/scala/io/gatling/core/session/el/ElCompiler.scala

```diff
@@ -395,7 +432,16 @@ final class ElCompiler private extends RegexParsers {
 
   private def randomUuid: Parser[ElPart[Any]] = "randomUuid()" ^^ (_ => RandomUUID)
 
-  private def nonSessionObject: Parser[ElPart[Any]] = currentTimeMillis | currentDate | randomUuid | randomSecureUuid
+  private def randomInt: Parser[ElPart[Any]] = "randomInt()" ^^ (_ => RandomInt)
+
+  private def randomIntRange: Parser[ElPart[Any]] = "randomInt(" ~> RangeRegex <~ ")" ^^ (n => RandomIntRange(n))
```

Comment:
Could you please try to perform the full parsing here instead of parsing the range in a second step?